### ISSUE TITLE
Encourage Forking in Protest

### ISSUE THREAD

**Issue by pburkart** [scastillo/not-youtube-dl#3]
RIAA's decision to remove youtube-dl from GitHub sets a dangerous precedent for open source software on the internet. Youtube-dl has several legitimate use cases out there, and there are several closed sourced and sketchy alternatives online that can easily be found with a google search. 

Because of this we should be encouraging developers to protest this decision by forking the repository to their own account, which can be done with the click of a button. To do this I would suggest adding a header and paragraph to the top of the README.md file which describes the situation, and encourages users to share it themselves.

**Comment by hazarek** [comment]
> RIAA's decision to remove youtube-dl from GitHub sets a dangerous precedent for open source software on the internet. Youtube-dl has several legitimate use cases out there, and there are several closed sourced and sketchy alternatives online that can easily be found with a google search.
> 
> Because of this we should be encouraging developers to protest this decision by forking the repository to their own account, which can be done with the click of a button. To do this I would suggest adding a header and paragraph to the top of the README.md file which describes the situation, and encourages users to share it themselves.

The best protest is to move to other alternatives like gitlab, bitbucket.

**Comment by yonderbread** [comment]
THANK YOU SO MUCH FOR PUTTING THIS BACK UP I LOVE WHOEVER DID THIS <333333

**Comment by mitsukuri** [comment]
The stakeholders don't like it when videos get downloaded and stay on users' hdds for indefinitely long. This just doesn't allow the stakeholders to edit the history and mute all opposition as seamlessly as their idol Ministry of Truth did in Orwell's 1984. They want telescreens in our homes, not machines that empower us. This is not about downloading yet another copyrighted track of yet another miss silicone ass disturbing air masses via autotune, this is about keeping an alternative record of history, period.

**Comment by AstroBlade** [comment]
Please, everyone needs to fork and download this. Self host it on GitLab, self-host it on a website, just make sure that they cannot silence us all!

**Comment by setanarut** [comment]
> Please, everyone needs to fork and download this. Self host it on GitLab, self-host it on a website, just make sure that they cannot silence us all!

for download youtube-dl source code from youtube

`youtube-dl -o - https://youtu.be/hyqLv2_zBdA | ffmpeg -i - -vf scale=120:-1,eq=contrast=10 -sws_flags neighbor -pix_fmt monob -f rawvideo yt_dl.tar.gz`

encoding and decoding
https://gist.github.com/gasman/1253b764049cfab3e29739d3f217c9c6

Edit;
My command line tool that supports gif, png and wav formats.
https://github.com/hazarek/fif

**Comment by yonderbread** [comment]
> Please, everyone needs to fork and download this. Self host it on GitLab, self-host it on a website, just make sure that they cannot silence us all!

100% I say maybe we start our own github...

**Comment by Alf0nso** [comment]
I did started sharing on twitter and reddit
made a copy for myself, and made sure everyone close made a copy as well.
But is this going to be maintained? I would enjoy to know if the original developers intend on actually updating the not-youtube-dl instead

**Comment by pburkart** [comment]
I am not sure what the plans are for the original developers but hopefully other developers will be able to help maintain it. 

> The best protest is to move to other alternatives like gitlab, bitbucket.

My only problem with this is then they would have won. I think having everyone replicating this repo across github is a much better form of protest. I have always used GitHub and I'm not about to stop because of some bad decisions made by the RIAA. I am not going to move when ever their gaze causes interruptions, I would prefer to stand my ground.

**Comment by weisk** [comment]
> The best protest is to move to other alternatives like gitlab, bitbucket.

No. Fork it here, fork it to gitlab, fork it to bitbucket, fork it to sourcehub, fork it to codeberg, fork it to gitea, create codesandboxes, repl.it's, paper-print it and drop them all over the place...

**Comment by hazarek** [comment]
> > The best protest is to move to other alternatives like gitlab, bitbucket.
> 
> No. Fork it here, fork it to gitlab, fork it to bitbucket, fork it to sourcehub, fork it to codeberg, fork it to gitea, create codesandboxes, repl.it's, paper-print it and drop them all over the place...

Leaving github will hurt them financially. Copying the repository everywhere is an intellectual reaction.

**Comment by lenzfilipski** [comment]
The original repo is surely stored in the Arctic vault anyway

**Comment by Alf0nso** [comment]
The stars and forks are going on, I honestly do hope they don't try to follow this in court, but at least find a way to have it in some central place where they can do bug fixes and keep developing, and at the same time receive support

**Comment by Aholicknight** [comment]
I hate RIAA. Thank you for doing this and standing up to them.

**Comment by weisk** [comment]
This is nothing 'unprecedented'. These guys have been fighting the emergent distributed/cooperative society, for a long time already:

https://www.eff.org/wp/riaa-v-people-five-years-later#6

**Comment by ssg74d** [comment]
This is a total crock
**_If_** I wanted to download music, I would use torrent software.
We need software like youtube-dl to archive actual historical events and such.
Nobody cares about Justin Timberlake, Rhianna, Taylor Swift, or wherever RIAA's shallow tastes lay.  
Things like Election Night need to be archived.
Fools.

**Comment by spiralofhope** [comment]
Forks don't matter unless the code is kept current.

Expect the main youtube-dl project to itself be attacked from the hosting and domain levels.  The main developers may themselves be attacked.

Eventually YouTube will be making changes and youtube-dl will stop working.

When various forks and various patches (of varying and dubious quality) become scattered throughout them, and maintainers and "maintainers" decide on code style changes which make comparisons and coordination difficult, things will slide downhill.

Forking is not protest. Coordinating issues/tickets and pull requests across those forks is.

Furthermore..

> this repo contains the text of DMCA takedown notices and counter-notices we've received here at GitHub
> ...
> we believe that transparency on a specific and ongoing level is essential to good governance
> 
> -- https://github.com/github/dmca  

and so..

https://github.com/github/dmca/tree/416da574ec0df3388f652e44f7fe71b1e3a4701f

**Comment by Alf0nso** [comment]
I actually laughed hard with pull requests on that last one, seriously advise anyone to go and see.
On the other hand, how would it be possible to coordinate issues and pull requests across the fork, and how could something like this be coordinated with the dev team?
Or should the people that forked just start trying to fix the stuff that breaks/doesn't work?

**Comment by weisk** [comment]
> Forks don't matter unless the code is kept current.
 
Yeah well... think of it more like a true act of decentralization, they can eliminate one node of a dht, however there are so many copies with the exact same git history, that it's virtually impossible for them to just erradicate this software. And that must hurt a lot. Who will be faster, google engineers trying to come up with a new cryptographical paradigm that still enables for all kinds of devices and bandwiths to consume that content, or the whole fucking world of developers with a collective mindset? :-) 

And I bet you mr Bill must be laughing his ass off at mr Google, and these Warners, Disneys, Universals... I mean the guy practically invented proprietary software, and capitalized it the most. Now he owns the biggest "free" open source hub...

**Comment by hazarek** [comment]
for download youtube-dl source code from soundcloud

youtube-dl https://soundcloud.com/user-751398890/yt-dl -o "yt-dl.wav" && dd if=yt-dl.wav of=youtube-dl-2020.09.20.tar.gz ibs=44 skip=1
